tests/servers/tests.py

```diff
@@ -27,6 +29,34 @@
 }
 
 
+class LiveServerThreadTest(TestCase):
+
+    def run_live_server_thread(self, connections_override=None):
+        thread = LiveServerTestCase._create_server_thread(connections_override)
+
+        thread.daemon = True
+        thread.start()
+        thread.is_ready.wait()
+        thread.terminate()
+        thread.join()
+
+    @unittest.skipIf(connection.vendor == 'sqlite', "sqlite does not permit closing connections")
```

Comment:
> I guess this is truly sqlite specific such that no third-party database backend would need to skip the test? It seems a little confusing because the connections.close_all() line does execute under SQLite, right?

@timgraham Yes, it executes. The reason it's being skipped is that the SQLite backend's `close()` method has special-case logic to make it a no-op when it's an in-memory db. See [here](https://github.com/django/django/blob/c969b17ad83e8efdd7f887b61ad75370395434ca/django/db/backends/sqlite3/base.py#L231) for the code. Thus, the test assertion to check that the connection gets closed will (and does) fail. The connection doesn't get closed because `close()` in that case doesn't do anything.

Technically, I guess we really only need to skip if it's SQLite _and_ an in-memory db. I can make that change.

@charettes Closing only the non-overridden connections wouldn't help because the issue is that the `close()` method is a no-op with when SQLite is being used with an in-memory database. So the test to check that the connection is closed will always fail (there's nothing to test).

I'm also using the overridden connections argument in an essential way in the test. It's how I'm passing the connection to the thread, so the main thread is later able to check that it's closed. It's okay to close the overridden connection because the only purpose of `connections_override` is to give `LiveServerTestCase` a way to pass the connection to the thread when in-memory SQLite is being used (which matches when `close()` is a no-op). See [here](https://github.com/django/django/blob/c969b17ad83e8efdd7f887b61ad75370395434ca/django/test/testcases.py#L1296) for that code.